### PR TITLE
Fix script injection during build

### LINKED ISSUE #12171: Shared entrypoint in injectRoute() Removes Client-Side Scripts in SSR Build
### Astro Info

```block
Astro                    v4.15.12
Node                     v18.20.2
System                   macOS (arm64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/node
Integrations             routing
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

**Description:** In a project using two injected routes with the same entrypoint, the client-side script is missing from one route after building the project, preventing dynamic behavior (e.g., a clock) from working.

**Steps to Reproduce:**
1. Define two injected routes in `astro.config.mjs` with the same `entrypoint`:
    ```js
    injectRoute({
      pattern: '/hello',
      entrypoint: './src/routes/template.astro'
    });

    injectRoute({
      pattern: '/blog/[...slug]',
      entrypoint: './src/routes/template.astro'
    });
    ```
2. Run the project in dev mode using `npm run dev`.
3. Navigate to http://localhost:4321/hello and http://localhost:4321/blog/welcome. Both routes correctly display a dynamic clock.
4. Build and serve the project using `npm run build && npm run server`.
5. Navigate to the same routes.

**Expected Behavior:**
Both `/hello` and `/blog/welcome` should display a dynamic clock.

**Actual Behavior:**
- `/hello` displays the clock as expected.
- `/blog/welcome` does **not** display the clock. The necessary `<script>` is missing from the page's source.

**Workaround:**
Removing the `/hello` injected route fixes the issue, and the clock works as expected on `/blog/` URLs after the build.

**Relevant Sources:**
- `view-source:http://localhost:4321/hello`
    ```html
    <!DOCTYPE html><script>window.localserviceSettings = {};</script><script src="/.localservice@runtime.134daa3c.js"></script><script type="module">document.getElementById("time");setInterval(()=>{time.innerText=new Date().toISOString()},100);</script><a href="/">back home</a> <h1>/hello</h1> <span id="time"></span>
    ```
- `view-source:http://localhost:4321/blog/welcome`
    ```html
    <!DOCTYPE html><a href="/">back home</a> <h1>/blog/welcome</h1> <span id="time"></span>
    ```

### What's the expected result?

All routes using the same entrypoint should include the necessary client-side scripts and behave consistently in both development and production builds.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-csdu7e?file=README.md

### Participation

- [x] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by apatel369**
## Changes
Closes #12171 
This pull request addresses issue #12171 where shared entrypoints in `injectRoute()` remove client-side scripts in SSR builds. The changes ensure that client-side scripts are correctly included for all routes using the same entrypoint.

## Testing
added tests 

## Docs
Internal bug fix so docs update not needed

### CODE REVIEW

**Comment by ematipico** [comment]
Should we add new tests?

**Comment by apatel369** [comment]
> Should we add new tests?

Added. Thanks for your patience. Let me know if any changes are needed.

File: .changeset/slimy-pets-lick.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fix issue where shared entrypoints in injectRoute() remove client-side scripts in SSR builds.
```
Comment (ematipico) [inline .changeset/slimy-pets-lick.md:5]:
```suggestion
Fixes an issue where scripts were not correctly injected during the build. The issue was triggered when there were injected routes with the same `entrypoint` and different `pattern`
```

File: packages/astro/src/core/build/plugins/plugin-hoisted-scripts.ts [hunk 2]
```diff
@@ -73,23 +72,18 @@ export function vitePluginHoistedScripts(
...
-								value: output.code,
-							};
-							removeFromBundle = true;
-						} else {
-							pageInfo.hoistedScript = {
-								type: 'external',
-								value: id,
-							};
-						}
+				for (const pageData of getPageDatasByHoistedScriptId(internals, facadeId)) {
```
Comment (ematipico) [inline packages/astro/src/core/build/plugins/plugin-hoisted-scripts.ts:75]: As you noticed from the Github bot comments, `settings` isn't used anymore, so we can remove it

Comment (apatel369) [reply packages/astro/src/core/build/plugins/plugin-hoisted-scripts.ts:73]: Removed. Thanks.

**Comment by ematipico** [comment]
@apatel369 the build is failing, can you please fix it?

**Comment by apatel369** [comment]
> @apatel369 the build is failing, can you please fix it?

Fixed.

**Review by ematipico — APPROVED** [review]

**Pull request merged**